core/node/libp2p/transport.go

```diff
@@ -31,15 +32,24 @@ func Transports(tptConfig config.Transports) interface{} {
 
 		if tptConfig.Network.QUIC.WithDefault(!privateNetworkEnabled) {
 			if privateNetworkEnabled {
-				// QUIC was force enabled while the private network was turned on.
-				// Fail and tell the user.
 				return opts, fmt.Errorf(
 					"The QUIC transport does not support private networks. " +
 						"Please disable Swarm.Transports.Network.QUIC.",
 				)
 			}
 			// TODO(9290): Make WithMetrics configurable
-			opts.Opts = append(opts.Opts, libp2p.Transport(libp2pquic.NewTransport, libp2pquic.WithMetrics()))
+			opts.Opts = append(opts.Opts, libp2p.Transport(quic.NewTransport, quic.WithMetrics()))
+		}
+
+		// TODO(9292): Remove the false && to allows it enabled by default
+		if tptConfig.Network.WebTransport.WithDefault(false && !privateNetworkEnabled) {
+			if privateNetworkEnabled {
```

Comment:
You can remove this check.

Reply:
I cannot.
Look at how quic is implemented above.

FYI the point of this is to catch someone that explicitelly add `"WebTransport": true` in their config while having pnet enabled.

Reply:
@ajnavarro I cannot,
What you are proposing turn using explicitly enabling webtransport in the config while pnet is in use into a silent error (webtransport isn't being enabled even tho you set `WebTransport: true`).

If you have incompatible config options we want to fail loudly so you go fix them.
We don't want to silently disable features you think are enabled.

Reply:
The current QUIC behaviour (which I want to keep):
|                      | PNET Off                     | PNET On                      |
|----------------------|------------------------------|------------------------------|
| WebTransport Off     | Works, WebTransport Disabled | Works, WebTransport Disabled |
| WebTransport Default | Works, WebTransport Enabled  | Works, WebTransport Disabled |
| WebTransport On      | Works, WebTransport Enabled  | Errors                       |

What you are proposing:
|                      | PNET Off                     | PNET On                      |
|----------------------|------------------------------|------------------------------|
| WebTransport Off     | Works, WebTransport Disabled | Works, WebTransport Disabled |
| WebTransport Default | Works, WebTransport Enabled  | Works, WebTransport Disabled |
| WebTransport On      | Works, WebTransport Enabled  | Works, WebTransport Disabled |